features/google/src/model/analyses/ministories/awakening-location-analysis.js

```diff
@@ -0,0 +1,118 @@
+import { RootAnalysis } from "@polypoly-eu/poly-analysis";
+import analysisKeys from "../analysisKeys";
+import moment from "moment";
+
+const MIN_MORNING = 5;
+const MAX_MORNING = 10;
+const SLEEP_DURATION = 5;
+
+export default class AwakeningLocationAnalysis extends RootAnalysis {
+    convertToDateString(date) {
+        return `${
+            date.getDate() < 10 ? `0${date.getDate()}` : date.getDate()
+        }/${
+            date.getUTCMonth() + 1 < 10
+                ? `0${date.getUTCMonth() + 1}`
+                : date.getUTCMonth() + 1
+        }/${date.getFullYear()}`;
+    }
+    containsWakingHours(startDate, endDate, daysDiff) {
+        const startHour = startDate.getHours();
+        const endHour = endDate.getHours();
+        //TODO currently, the placeVisits that span more than 1 day automatically
+        //become the awaking place of both days even if the last time-stamp is at 2:00
+        if (daysDiff > 1) return true;
+        if (endHour <= MIN_MORNING) {
+            return false;
+        }
+        if (daysDiff == 0) {
+            if (startHour <= MIN_MORNING && endHour <= MIN_MORNING)
+                return false;
+            if (startHour >= MIN_MORNING) return false;
+        }
+
+        return true;
+    }
+
+    getDateDiff(startDate, endDate) {
+        const momentStart = moment(startDate).startOf("day");
```

Comment:
I have been checking out where you use this module, and [the documentation itself](https://momentjs.com/docs/) states that it's a bit heavy for bundling (old design, tree-shaking does not work) and recomends others like [Luxon](https://github.com/moment/luxon/). I'm looking at the code here in the second case:

https://github.com/moment/luxon/blob/master/src/datetime.js#L1438-L1478

and it looks like it's using a very simple heuristic to return "0" for day; essentially it's looking at quantities and whether they start with 1 (days or months) or 0 (hours, minutes)

moment.js seems to follow the same logic:

https://github.com/moment/luxon/blob/master/src/datetime.js#L1438-L1478

So, all in all, I'm not totally sure we want to add this new dependency (or, for that matter, that it does what you want)

Reply:
yeah that's true, I just removed it. I guess we might need it later but for the time being I think we should be fine without it